### PR TITLE
[webview_flutter_platform_interface] Adds option to override console log

### PR DESCRIPTION
**Pull request by mvanbeusekom**
Adds an option to the `webview_flutter_platform_interface` to register a JavaScript console callback. This will allow developers to receive JavaScript console messages in a Dart callback.

This PR contains the `webview_flutter_platform_interface` changes from PR #4541.

Related issue: flutter/flutter#32908

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/webview_flutter/webview_flutter_platform_interface/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## 2.5.0
+
+* Adds support to register a callback to intercept messages that are written to
+  the JavaScript console. See `PlatformWebViewController.setConsoleLogCallback`.
```
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_platform_interface/CHANGELOG.md:4]:
```suggestion
  the JavaScript console. See `PlatformWebViewController.setOnConsoleMessage`.
```

File: packages/webview_flutter/webview_flutter_platform_interface/test/platform_webview_controller_test.dart [hunk 2]
```diff
@@ -387,6 +387,20 @@ void main() {
       throwsUnimplementedError,
     );
   });
+
+  test(
+      'Default implementation of setConsoleLogCallback should throw unimplemented error',
```
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_platform_interface/test/platform_webview_controller_test.dart:392]:
```suggestion
      'Default implementation of setOnConsoleMessage should throw unimplemented error',
```

**Review by bparrishMines — APPROVED** [review]
LGTM

File: packages/webview_flutter/webview_flutter_platform_interface/lib/src/platform_webview_controller.dart [hunk 3]
```diff
@@ -270,6 +270,15 @@ abstract class PlatformWebViewController extends PlatformInterface {
       'setOnPermissionRequest is not implemented on the current platform',
     );
   }
+
+  /// Sets a callback that notifies the host application of any console messages
+  /// written to the JavaScript console.
+  Future<void> setOnConsoleMessage(
+      void Function(JavaScriptConsoleMessage consoleMessage) onConsoleMessage) {
+    throw UnimplementedError(
+      'setConsoleLogCallback is not implemented on the current platform',
```
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_platform_interface/lib/src/platform_webview_controller.dart:279]:
```suggestion
      'setOnConsoleMessage is not implemented on the current platform',
```

**Review by bparrishMines — APPROVED** [review]
LGTM

**Review by stuartmorgan — APPROVED** [review]
LGTM

@bparrishMines Do you want to apply your suggested changes and submit?

**Pull request merged**